octopus/src/main/scala/octopus/Validator.scala

```diff
@@ -10,6 +10,8 @@ import scala.util.control.NonFatal
 trait Validator[T] {
 
   def validate(obj: T): List[ValidationError]
+
+  def comap[U](f: U => T): Validator[U] = (value: U) => validate(f(value))
```

Comment:
I'd prefer to be consistent with current design, having type class purely simple and other operations on `Validator` defined in `dsl.ValidatorOps`. Can you move `comap` there? Beside this, LGTM.